src/main/java/fence/Fence.java

```diff
@@ -0,0 +1,86 @@
+package fence;
```

Comment:
Your error handling with custom exceptions (e.g., NoSuchElementException, DateTimeParseException) is effective.
Consider adding more specific error messages to guide the user on what they did wrong. This can enhance the user experience.